param_passing/tutorial.md

```diff
@@ -0,0 +1,561 @@
+the attribute an empty value (e.g., `//pose[@relative_to=""]`).
+The `@name` attribute can only be modified when the `modify` action is defined in the `@element_id` element.
+
+#### Modify examples
+
+**Example 1**
+
+To `modify` a name attribute only, in this case the original model has `//visual[@name="visual"]`:
+
+```xml
+<!-- //experimental:params -->
+<visual element_id="path::to::visual" name="new_name" action="modify"/>
+```
+
+this only modifies the `@name` of the `//visual` to be `new_name`.
+There are no other listed attributes or child elements so nothing else is changed.
+
+To `modify` child elements and not the name attribute provide
+the full path to the element to be modified with the updated value:
+
+```xml
+<!-- //experimental:params -->
+<visual element_id="path::to::visual" action="modify">
+  <geometry>
+    <sphere>
+      <radius>1.0</radius>
+    </sphere>
+  </geometry>
+</visual>
+```
+
+**Example 2**
+
+Looking at an `//inertial` example, if the original model has:
+
+```xml
+<!-- included model -->
+...
+<link name="chassis">
+  <pose relative_to="some_frame">1 0 0 0 0 0</pose>
+  <inertial>
+    <mass>1.14395</mass>
+    <inertia>
+      <ixx>0.126164</ixx>
+      <ixy>0</ixy>
+      <ixz>0</ixz>
+      <iyy>0.416519</iyy>
+      <iyz>0</iyz>
+      <izz>0.481014</izz>
+    </inertia>
+  </inertial>
+  ...
+</link>
+...
+```
+
+To `modify` the `@relative_to`, `//mass`, and some `//inertia` elements:
+
+```xml
+<!-- //experimental:params -->
+<link element_id="path::to::chassis" action="modify">
+  <pose relative_to="new_frame">1 0 0 0 0 0</pose>
+  <inertial>
+    <mass>2.637</mass>
+    <inertia>
+      <ixx>0.02467</ixx>
+      <iyy>0.04411</iyy>
+      <izz>0.02467</izz>
+    </inertia>
+  </inertial>
+</link>
+```
+
+Note that `//pose` values are the same values as the original.
+Currently, there is no way to specify if element values should remain the same
+so they must be provided (this can only be done with attributes by not providing them).
+
+The expected updated model would be:
+
+```xml
+<!-- updated included model -->
+...
+<link name="chassis">
+  <pose relative_to="new_frame">1 0 0 0 0 0</pose>
+  <inertial>
+    <mass>2.637</mass>
+    <inertia>
+      <ixx>0.02467</ixx>
+      <ixy>0</ixy>
+      <ixz>0</ixz>
+      <iyy>0.04411</iyy>
+      <iyz>0</iyz>
+      <izz>0.02467</izz>
+    </inertia>
+  </inertial>
+  ...
+</link>
+...
+```
+
+---
+
+### Remove action
+
+The `remove` action removes elements from the original model.
+
+When `remove` is declared where `@element_id` is specified, the entire element is removed.
+Any listed children elements are ignored.
+When `remove` is declared in the direct children of `@element_id` there are 2 cases to consider.
+If the direct child (where `remove` is defined):
+1. Has no children elements, the entire element is removed
+1. Has children, the individual children are removed but not the element where `remove` is defined
```

Comment:
refer to one of the examples below? e.g. `see Example x`

Reply:
215e07b977d9033233462127d373fdbb6f19013a